common/models/user.js

```diff
@@ -382,14 +396,114 @@ module.exports = function(User) {
     }
   );
 
+  User.requestAuthLink = function requestAuthLink(email) {
+    if (!isEmail(email)) {
+      return Promise.reject(
+        new Error('The submitted email not valid.')
+        );
+    }
+
+    var userObj = {
+      username: 'fcc' + uuid.v4().slice(0, 8),
+      email: email,
+      emailVerified: false
+    };
+    return User.findOrCreate$({ where: { email: userObj.email }}, userObj)
+      .map(([ err, user, isCreated ]) => {
+        if (err) {
+          return dedent`
+           Oops, something is not right, please try again later.`;
+        }
+
+        if (!isDev) {
+          const minutesLeft = getWaitPeriod(user.emailAuthLinkTTL);
+          if (minutesLeft) {
+            const timeToWait = minutesLeft ?
+              `${minutesLeft} minute${minutesLeft > 1 ? 's' : ''}` :
+              'a few seconds';
+            debug('request before wait time : ' + timeToWait);
+            return dedent`
+              Please wait ${timeToWait} to resend an authentication link.`;
+          }
+        }
+
+        let emailTemplate = 'user-request-sign-in.ejs';
+        if (isCreated) {
+          emailTemplate = 'user-request-sign-up.ejs';
+        }
+
+        // create a temporary access token with ttl for 1 hour
+        user.createAccessToken({ ttl: 60 * 60 * 1000 }, (err, token) => {
+          if (err) { throw err; }
+
+          const { id: loginToken } = token;
+          const loginEmail = user.email;
+          const renderAuthEmail = loopback.template(path.join(
```

Comment:
Instead of creating this renderer per request, can we move this up into the module scope and then use the ternary above to select the appropriate renderer instead of just the template name?